snooty/util.py

```diff
@@ -648,3 +649,91 @@ def parse_toml_and_add_line_info(text: str) -> Dict[str, Any]:
             raise TOMLDecodeErrorWithSourceInfo(message, text.count("\n") + 1) from err
 
         raise err
+
+
+def deserialize_ast(
```

Comment:
I originally made this a `classmethod` of `n.Node` (see [commit](https://github.com/mongodb/snooty-parser/pull/638/commits/d024d2a1809dae58e01979ad964265feff202d4d)), but I ended up making a util method to leverage diagnostics. Otherwise, there was a circular dependency that I was having issues resolving.

Reply:
yeah, this is okay. The circular dependency issues around n.py are perennial -- SerializableType really almost needs to be in its own file, because everything wants access to it.